File: docs/WidgetMill/packages/spec/src/nodes/icon.ts
import { z } from "zod";
import { BindableColor } from "../scalars";

/** Icon — spec §2.7. `icon` (nome do catálogo Material) é obrigatório. */
export const IconNode = z.object({
  type: z.literal("icon"),
  props: z.object({
    icon: z.string(),
    size: z.number().optional(),
    color: BindableColor.optional(),
  }),
});

File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/icon.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../parsing/material_icons.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// Icon — spec §2.7 (catálogo curado de Material Icons).
Widget buildIcon(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return Icon(
    iconDataFrom(p['icon']),
    size: parseDouble(p['size']) ?? 24,
    color: parseColor(p['color']),
  );
}
